main.go

```diff
@@ -607,6 +608,63 @@ func main() {
 					)
 				})
 			}
+
+			// Traces.
+			if cfg.traces.enabled {
+				r.Group(func(r chi.Router) {
+					r.Use(authentication.WithTenantMiddlewares(pm.Middlewares))
+					r.Use(authentication.WithTenantHeader(cfg.traces.tenantHeader, tenantIDs))
+
+					// V2 query API, protected by RBAC
+					r.Mount("/api/traces/v1/{tenant}/api",
+						stripTenantPrefix("/api/traces/v1",
+							tracesv1.NewV2APIHandler(
+								cfg.traces.readEndpoint,
+								tracesv1.Logger(logger),
+								tracesv1.WithRegistry(reg),
+								tracesv1.WithHandlerInstrumenter(ins),
+								tracesv1.WithSpanRoutePrefix("/api/traces/v1/{tenant}"),
+								tracesv1.WithReadMiddleware(authorization.WithAuthorizers(authorizers, rbac.Read, "traces")),
+								tracesv1.WithReadMiddleware(logsv1.WithEnforceAuthorizationLabels()),
+								tracesv1.WithWriteMiddleware(authorization.WithAuthorizers(authorizers, rbac.Write, "traces")),
+							),
+						),
+					)
+
+					// Static UI (.js and .css), not protected by RBAC -- any user can see UI
```

Comment:
I'm wondering if there's any reason we do not want to enforce authorization specifically on UI?

Also I'm wondering if following the convention we have with other signals would make sense here as well, i.e. having everything organize under one handler instead of splitting it into API, UI etc.

Reply:
If I go to http://localhost:8080/api/traces/v1/test-oidc/search/ I am redirected to http://localhost:5556/dex/auth/local/login?back=&state=vcqh2yns46maemaiyfh2k33xm where I am presented with a Dex login screen.  So we do enforce **authentication** through the `r.Use(authentication.WithAccessToken())` earlier in the function.

I felt like the RBAC permission for traces for for reading and writing traces, not UI.  By not doing RBAC the UI comes up the UI tries to fetch trace "service names", and the user sees the Unauthorized message within a nice-looking Jaeger UI.  Do you want me to add RBAC here to block the UI from coming up at all?

I originally used a single handler until I chose to use `httputil.ReverseProxy.ModifyResponse`.  I didn't want that logic to run over the /static resources of the Jaeger UI, only the generated HTML.  I split into two handlers to make that logic more clear.

Reply:
By not doing RBAC for the UI resources we allow the UI to load and make a failing API query for traces.

If I remove RBAC permission for traces the UI for http://localhost:8080/api/traces/v1/test-oidc/search/ looks like this:

<img width="885" alt="no-trace-rbac" src="https://user-images.githubusercontent.com/3237651/159535134-000996d2-8593-4837-a11c-869a010553a9.png">

I thought it was cute to allow it to work this way.  I am happy to secure the UI resources behind RBAC permission to read traces if that is your request.

Reply:
I'm just trying to think what's the most reasonable solution here, but I'm open to other opinions. I'm currently leaning more towards authorizing on all endpoints, since if the tenants is not authorized to read / write trace signals, there's no reason for the tenant to even be accessing the UI.

Reply:
I think it makes sense what @matej-g proposes - also it may avoid unnecessary loading of the UI + trial to fetch traces. So maybe is better to consistently to add authorization on all endpoints